### ISSUE TITLE
Would patreon integration ever be possible? :O

### ISSUE THREAD

**Issue by captainharrie** [Ayemae/Grawlix-Webcomic-CMS#24]
I'm not fully certain on how scheduled posts go from inaccessible to accessible currently (does grawlix check if the publish time is less than or equal to the current time before building the page..?), and there's obviously no authentication happening at the moment because you can't access said scheduled posts even as the admin,

but I recall back on smackjeeves they had patreon integration which meant that on a scheduled page if you set an early access date and which tier unlocks it, when a reader subscribed to that tier authenticated through the patreon API they could read those scheduled pages early.

there'd obviously need to be a change to how the scheduled pages work so if theyre not yet live you can view them if youre authorised to before any kind of patreon integration could happen, but I do think that would be useful in and of itself! But I've no idea how feasible that first step would be it sounds like a pretty major overhaul to me. but perhaps it could be a future plan idk!! :o

**Comment by eishiya** [comment]
Grawlix just checks the current date versus the publication date, yeah. It's doable to modify this check to include a patron grace period, as it's effectively just subtracting the appropriate grace period from the publication date when doing the check. There would need to be a page in the admin panel to define per-tier grace periods (with the attendant database schema changes to store them and the Patreon API key), but actually comparing against these should be simple enough. The hardest part would be implementing the authentication against Patreon's API, but I don't think there's anything in Grawlix that would make it impossible.

Unfortunately, at the rate Patreon's been going lately, I worry it'll be practically dead before I or Ayemae can get around to implementing this ):

**Comment by captainharrie** [comment]
Yeah I did kind of want to suggest Ko-Fi at first since thats the most popular patreon alternative currently, but the extent of their "API" is a webhook for notifications of sales n subscriptions 😫 maybe theyll release a proper api later down the line... there are other alteratives though, [buymeacoffee has an api](https://developers.buymeacoffee.com/)! So if patreon specifically doesnt shake out, I should think there would be another platform for monthly donations out there that people are using. I just reckon it'd encourage people to subscribe more if the benefits were directly on the website itself :o
